src/main/scala/fi/vm/sade/vst/server/Server.scala

```diff
@@ -26,8 +26,25 @@ class Server(routes: Routes, config: ServerConfig, implicit val system: ActorSys
 
   def start(): Unit = {
     System.setProperty("logback.debug", "true")
+    assertNoSlf4jLog4jOnClasspath()
     val handler = Http().bindAndHandle(route, "0.0.0.0", port)
     logger.info(s"Starting server on port $port")
     handler.failed.foreach { case ex: Exception => logger.error("Failed to bind to port", ex) }
   }
+
+  private def assertNoSlf4jLog4jOnClasspath(): Unit = {
+    assertNotOnClassPath("org.slf4j.impl.Log4jLoggerFactory", "org.apache.log4j.Logger")
+  }
+
+  def assertNotOnClassPath(classNames: String*): Unit = {
+    classNames.foreach { fullyQualifiedName =>
+      try {
+        Class.forName(fullyQualifiedName)
+        throw new IllegalStateException(s"${getClass.getName} found class $fullyQualifiedName on the classpath. " +
+          s"This might break logback logging. Please remove log4j from and its bindings from the classpath.")
```

Comment:
Tämä on aika nyansseja, mutta on pikkuisen outoa että tämän yleisemmän funktion sisällä puhutaan logituksesta. Sitä voisi ehkä joku käyttää muuhunkin kuin juuri noiden logitusluokkien tarkistamiseen. Tän huomion voisi siirtää tuon assertNoSlf4jLog4jOnClasspath:in alle.